### PR TITLE
feat: Allow user to add service account labels

### PR DESCRIPTION
**Pull request by zopanix**
#### PR Description

Allow helm chart users to add service account labels

This can be useful to use OIDC required labels for grafana-agent to authenticate to services.

For example, Azure Workload Identity requires a labels to set additional labels on the service account for it to function properly.

#### Which issue(s) this PR fixes

n/a

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Comment by zopanix** [comment]
Just to say I signed the CLA, should get updated soon enough

File: operations/helm/charts/grafana-agent/Chart.yaml
Comment (rfratto) [inline operations/helm/charts/grafana-agent/Chart.yaml:5]: Hi, can you roll this change back? We'll increment the version when the Helm chart is ready for a release; there's a small process to follow so I don't want to tie up the release with this PR.

File: operations/helm/charts/grafana-agent/CHANGELOG.md
Comment (rfratto) [inline operations/helm/charts/grafana-agent/CHANGELOG.md:17]: I'd like to see the name of the new field in the changelog entry: 

```suggestion
- Add `serviceAccount.additionalLabels` to values.yaml to enable setting additional labels on the created service account.
```

File: operations/helm/charts/grafana-agent/Chart.yaml
Comment (captncraig) [inline operations/helm/charts/grafana-agent/Chart.yaml:5]: Let's not increment the version quite yet. There is one other pending PR for helm, and then I'll make a new release shortly.

File: operations/helm/charts/grafana-agent/templates/serviceaccount.yaml
Comment (rfratto) [inline operations/helm/charts/grafana-agent/templates/serviceaccount.yaml:10]: Can you add a corresponding test values.yaml file for this in `operations/helm/charts/grafana-agent/ci` and run `make generate-helm-tests` to generate the manifests for it? That way we can have regression tests in case this breaks.

**Review by rfratto — COMMENTED** [review]
Hi! Thanks for the contribution. I have a few comments but LGTM after that.

**Comment by christophe-scalepad** [comment]
@rfratto: I've made the requested changes, I've also rebased on upstream since there were merge conflicts, should be good to go. Thanks for the feedback

**Review by captncraig — APPROVED** [review]

**Pull request merged**
